server/commons/notify.coffee

```diff
@@ -0,0 +1,36 @@
+slack = require '../slack'
+sendwithus = require '../sendwithus'
+User = require '../models/User'
+
+# TODO: Refactor notification (slack, watcher emails) logic here
+
+module.exports = 
+  notifyChangesMadeToDoc: (req, doc) ->
+    # TODO: Stop using headers to pass edit paths. Perhaps should be a method property for Mongoose models
+    editPath = req.headers['x-current-path']
+    docLink = "http://codecombat.com#{editPath}"
+
+    # Post a message on Slack
+    message = "#{req.user.get('name')} saved a change to #{doc.get('name')}: #{doc.get('commitMessage') or '(no commit message)'} #{docLink}"
+    slack.sendSlackMessage message, ['artisans']
+
+    # Send emails to watchers
+    watchers = doc.get('watchers') or []
+    # Don't send these emails to the person who submitted the patch, or to Nick, George, or Scott.
+    watchers = (w for w in watchers when not w.equals(req.user.get('_id')) and not (w + '' in ['512ef4805a67a8c507000001', '5162fab9c92b4c751e000274', '51538fdb812dd9af02000001']))
```

Comment:
Weird. In that case stick with this — using toString would be confusing given that on the console it includes the `ObjectId()` part.